builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,364 @@
+		return multistep.ActionHalt
+	}
+
+	logger.Info("Finished importing the image from %s to %s.", s.SourceURL, s.TargetLocationName)
+
+	return multistep.ActionContinue
+}
+
+func (s *StepImportImage) validate(ctx context.Context, logger *PackerLogger) error {
+	logger.Info("Validating image import request...")
+
+	var err error
+	if err = s.isImportFeatureEnabled(ctx, logger); err != nil {
+		return err
+	}
+
+	if err = s.isImportSourceValid(); err != nil {
+		return err
+	}
+
+	if err = s.isImportTargetValid(ctx, logger); err != nil {
+		return err
+	}
+
+	logger.Info("Image import request source and target are valid.")
+	return nil
+}
+
+func (s *StepImportImage) Cleanup(state multistep.StateBag) {
+	if state.Get(StateKeyImageImportRequestCreated) == false {
+		// Either the image import step was skipped or the object was not created successfully.
+		// Skip deleting the ContentLibraryItemImportRequest object.
+		return
+	}
+
+	logger := state.Get("logger").(*PackerLogger)
+	if s.KeepInputArtifact {
+		logger.Info("Skipping clean up of the ContentLibraryItemImportRequest object as specified in config.")
+		return
+	}
+
+	logger.Info(fmt.Sprintf("Deleting the ContentLibraryItemImportRequest object %s in namespace %s.",
+		s.ImportImageConfig.ImportRequestName, s.Namespace))
+	ctx := context.Background()
+	itemImportReqObj := &imgregv1.ContentLibraryItemImportRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ImportImageConfig.ImportRequestName,
+			Namespace: s.Namespace,
+		},
+	}
+	if err := s.KubeWatchClient.Delete(ctx, itemImportReqObj); err != nil {
+		logger.Error("error deleting the ContentLibraryItemImportRequest object : %s", err)
+	} else {
+		logger.Info("Successfully deleted the ContentLibraryItemImportRequest object %s in namespace %s.",
+			s.ImportImageConfig.ImportRequestName, s.Namespace)
+	}
+}
+
+func (s *StepImportImage) initStep(state multistep.StateBag, logger *PackerLogger) error {
+	if err := CheckRequiredStates(state,
+		StateKeySupervisorNamespace,
+		StateKeyKubeClient,
+	); err != nil {
+		logger.Error("error checking required states: %s", err)
+		return err
+	}
+
+	var ok bool
+	if s.Namespace, ok = state.Get(StateKeySupervisorNamespace).(string); !ok {
+		return fmt.Errorf("failed to cast %s to type string", StateKeySupervisorNamespace)
+	}
+	if s.KubeWatchClient, ok = state.Get(StateKeyKubeClient).(client.WithWatch); !ok {
+		return fmt.Errorf("failed to cast %s to type client.WithWatch", StateKeyKubeClient)
+	}
+
+	s.SourceURL = s.ImportImageConfig.ImportSourceURL
+	s.SourceSSLCertificate = s.ImportImageConfig.ImportSourceSSLCertificate
+	s.TargetLocationName = s.ImportImageConfig.ImportTargetLocationName
+	if s.ImportImageConfig.ImportTargetImageType != "" {
+		s.TargetItemType = imgregv1.ContentLibraryItemType(strings.ToUpper(s.ImportImageConfig.ImportTargetImageType))
+	}
+
+	// Use the ImageName from create source config as the target item name.
+	s.TargetItemName = s.CreateSourceConfig.ImageName
```

Comment:
`image_name` is required today, I feel it may be better to re-use it instead of introducing another optional config and making the `image_name` optional, since it is the same image that will be used to create the source VM, either imported or pre-created.

Reply:
I think the fewer required configs we have, the better experience would be for using this plugin. If the import source URL is provided, there is no need for users to specify the image name as well. I am fine if you want to reuse the `image_name` config here, but we should also update it to be an optional config. We could add a check in `config.Prepare()` to tigger an error if neither of the image name or import source URL is set.

Reply:
Yeah, the configs exposed to users are not really organized by steps, different steps can declare required configs and do pre-checks to make sure needed configs are provided, it may provide better UX with less configs, esp for configs that connect adjacent steps. Agreed, we better make this config optional if import image is configured.

Reply:
It turns out configs with the same name cannot be used directly except cross-reference, and this is optional step, so better not move configs here, adding an optional config instead.